Tests/External/Plugins/ASCompletion.Tests/Completion/ASGeneratorTests.cs

```diff
@@ -0,0 +1,611 @@
+            public void TextOnStartLine()
+            {
+                var sci = GetBaseScintillaControl();
+                sci.Text = "function test():void {trace(1);}";
+                sci.ConfigurationLanguage = "as3";
+                sci.Colourise(0, -1);
+                int funcBodyStart = ASGenerator.GetBodyStart(0, 2, sci);
+
+                Assert.AreEqual(25, funcBodyStart);
+                Assert.AreEqual("function test():void {\r\n\ttrace(1);}", sci.Text);
+            }
+
+            [Test]
+            public void BracketInCommentsOrText()
+            {
+                var sci = GetBaseScintillaControl();
+                sci.ConfigurationLanguage = "haxe";
+                sci.Text = "function test(arg:String='{', arg2:String=\"{\"):void/*{*/{\r\n}";
+                sci.Colourise(0, -1);
+                int funcBodyStart = ASGenerator.GetBodyStart(0, 1, sci);
+
+                Assert.AreEqual(60, funcBodyStart);
+            }
+
+            [Test]
+            public void MultiByteCharacters()
+            {
+                var sci = GetBaseScintillaControl();
+                sci.Text = "function test():void/*áéíóú*/\r\n{}";
+                sci.ConfigurationLanguage = "haxe";
+                sci.Colourise(0, -1);
+
+                int funcBodyStart = ASGenerator.GetBodyStart(0, 1, sci);
+
+                Assert.AreEqual(40, funcBodyStart);
+                Assert.AreEqual("function test():void/*áéíóú*/\r\n{\r\n\t\r\n}", sci.Text);
+            }
+
+            [Test]
+            [Ignore("Having only LineFrom and LineTo for members is not enough to handle these cases. FlashDevelop in general is not too kind when it comes to several members in the same line, but we could change the method to use positions and try to get the proper position before.")]
+            public void WithAnotherMemberInTheSameLine()
+            {
+                var sci = GetBaseScintillaControl();
+                sci.Text = "function tricky():void {} function test():void{\r\n\t\t\t}";
+                int funcBodyStart = ASGenerator.GetBodyStart(0, 1, sci);
+
+                Assert.AreEqual(49, funcBodyStart);
+            }
+
+
+            [Test]
+            public void BracketsInGenericConstraint()
+            {
+                var sci = GetBaseScintillaControl();
+                sci.ConfigurationLanguage = "haxe";
+                sci.Text = "function test<T:{}>(arg:T):void{\r\n\r\n}";
+                sci.Colourise(0, -1);
+                int funcBodyStart = ASGenerator.GetBodyStart(0, 1, sci);
+
+                Assert.AreEqual(34, funcBodyStart);
+                Assert.AreEqual("function test<T:{}>(arg:T):void{\r\n\r\n}", sci.Text);
+            }
+        }
+
+        public class GenerateJob : ASGeneratorTests
+        {
+            private ClassModel GetAs3ImplementInterfaceModel()
+            {
+                var interfaceModel = new ClassModel { InFile = new FileModel(), Name = "ITest", Type = "ITest" };
```

Comment:
I already thought of it a lot in the past, I know what you mean, but the thing is that then you don't have the desired control over the input, and there are higher chances of having problems.